addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4283,23 +4283,29 @@ export class OdooEditor extends EventTarget {
         // and the toolbar so we need to fix the selection to be based on the
         // editable children. Calling `getDeepRange` ensure the selection is
         // limited to the editable.
-        const containerSelector = '#wrap>*, .oe_structure>*, [contenteditable]';
-        const container =
-            (selection &&
-                closestElement(selection.anchorNode, containerSelector)) ||
-            // In case a suitable container could not be found then the
-            // selection is restricted inside the editable area.
-            this.editable;
-        if (
-            selection.anchorNode === container &&
-            selection.focusNode === container &&
-            selection.anchorOffset === 0 &&
-            selection.focusOffset === [...container.childNodes].length
-        ) {
-            getDeepRange(container, {select: true});
-            // The selection is changed in `getDeepRange` and will therefore
-            // re-trigger the _onSelectionChange.
-            return;
+        const isEmpty = !selection.anchorNode.childNodes.length;
+        // If the selection has no children, we must keep the current selection.
+        // Because if the selection is an empty editable structure, it will be
+        // selected again, creating an infine loop.
+        if (!isEmpty) {
```

Comment:
I'm reluctant to add a condition like this one on top of a piece of code that is supposed to be able to find the right container to select. Could you try adapting the selector below to add `:not(:empty)` to `.oe_structure` please? If it doesn't work then I suggest to move the `isEmpty` check to the condition of the `return` above, so that we return if the selection is not in the editable or if its container is empty. In that case I would rename `isEmpty` to `isContainerEmpty` and would also check that `anchorNode === focusNode`.

Reply:
@dmo-odoo Thanks for looking, I've made the change.

Reply:
Thanks, but I think you should you also check that `anchorNode === focusNode` otherwise your anchor can be in an empty node and the selection can span multiple nodes so that the focus ends up in a non-empty node, in which case `getDeepRange` should be called.

Reply:
@dmo-odoo isn't this already checked in the condition ?
```js
        if (
            selection.anchorNode === container && // <====== here
            selection.focusNode === container && // <====== here
            selection.anchorOffset === 0 &&
            selection.focusOffset === [...container.childNodes].length &&
            !isContainerEmpty
        ) {
```

Reply:
Well, yes and no, because it checks `container`, which actually may be a different node that `selection.anchorNode` for which you check the emptiness, because `selection.anchorNode` might be a `TextNode`, whereas container will always be an `Element`. Actually, now that I realize that, it would probably be better to replace `!isContainerEmpty` in the condition by `container.childNodes.length`. You can keep the comment above the check if you like.

Reply:
@dmo-odoo Thanks for looking, I've made the change.

Reply:
@dmo-odoo Thanks, I've made the change.